src/span_context.js

```diff
@@ -33,6 +33,7 @@ export default class SpanContext {
     _flags: number;
     _baggage: any;
     _debugId: ?string;
+    _samplingFinished: boolean;
```

Comment:
* I think this field deserves a big comment explaining its semantics and various scenarios. 
* per previous comment, s/finished/finalized/
* the sampling decision must become final as soon as the span context is injected into something, or used as a reference for another span (maybe not for all references, but certainly for childOf and followsFrom).